c2rust-transpile/src/lib.rs

```diff
@@ -225,12 +226,44 @@ fn get_module_name(
     file.to_str().map(String::from)
 }
 
+pub fn create_temp_compile_commands(sources: &[PathBuf]) -> PathBuf {
+    let temp_path = std::env::temp_dir().join("compile_commands.json");
+    let compile_commands: Vec<CompileCmd> = sources
+        .iter()
+        .map(|source_file| {
+            let absolute_path = fs::canonicalize(source_file)
+                .unwrap_or_else(|_| panic!("Could not canonicalize {}", source_file.display()));
+
+            CompileCmd {
+                directory: absolute_path.parent().unwrap().to_path_buf(),
```

Comment:
clang tooling's `CommonOptionsParser` (what we call), [uses `.` as the default directory when there's no compilation database](https://clang.llvm.org/doxygen/classclang_1_1tooling_1_1FixedCompilationDatabase.html#a1443b7812e6ffb5ea499c0e880de75fc), so we should probably match that.